packages/bot/src/components/flyout.jsx

```diff
@@ -7,98 +7,110 @@ import { config }           from '../scratch/help-content/help-content.config';
 import { connect }          from '../stores/connect';
 import                           '../assets/sass/scratch/flyout.scss';
 
-const Flyout = ({
-    is_help_content,
-    is_search_flyout,
-    block_nodes,
-    flyout_content,
-    flyout_width,
-    is_visible,
-    showHelpContent,
-}) => {
-    return (
-        <div
-            className={classNames(
-                'flyout', {
-                    'hidden'         : !is_visible,
-                    'flyout__search' : is_search_flyout,
-                    'flyout__help'   : is_help_content,
-                    'flyout__content': !is_help_content,
-                },
-            )
-            }
-            style={{ width: `${flyout_width}px` }}
-        >
-            {
-                is_help_content ?
-                    <HelpBase block_nodes={block_nodes} /> :
-                    Object.keys(flyout_content).map((key, index) => {
-                        const nodes = flyout_content[key];
-                        const node = nodes[0];
-                        const tag_name = node.tagName.toUpperCase();
+class Flyout extends React.PureComponent {
+    componentDidMount() {
+        this.props.onMount();
+    }
 
-                        switch (tag_name) {
-                            case Blockly.Xml.NODE_BLOCK: {
-                                const block_type = node.getAttribute('type');
+    componentWillUnmount() {
+        this.props.onUnmount();
+    }
```

Comment:
Flyout wouldn't be unMount when the flyout is hidden. Instead of removing the listener here, remove it in `Blockly.Toolbox.prototype.setSelectedItem` when `item === null`. Same goes to `onMount`

```
if(!item) {
 window.removeEventListener('click', ScratchStore.instance.flyout.clickOutsideFlyoutCallback)
} else {
 window.addEventListener('click', ScratchStore.instance.flyout.clickOutsideFlyoutCallback)
}
```

Also, remember to change back flyout component to functional component